File: src/main/java/no/fintlabs/mapping/selectable/ActiveTimePeriodSelectableMappingService.java
package no.fintlabs.mapping.selectable;

import no.fintlabs.model.Selectable;
import no.fintlabs.model.instance.ActiveTimePeriod;
import org.springframework.stereotype.Service;

@Service
public class ActiveTimePeriodSelectableMappingService {

    public Selectable<String> toSelectable(ActiveTimePeriod activeTimePeriod) {
        return Selectable
                .<String>builder()
                .value(activeTimePeriod.name())
                .label(getDisplayText(activeTimePeriod))
                .build();
    }

    private String getDisplayText(ActiveTimePeriod activeTimePeriod) {
        return switch (activeTimePeriod) {
            case TODAY -> "Denne dagen";
            case THIS_WEEK -> "Denne uka";
            case THIS_MONTH -> "Denne måneden";
            case THIS_YEAR -> "Dette året";
        };
    }

}

File: src/main/kotlin/no/novari/flyt/history/mapping/selectable/ActiveTimePeriodSelectableMappingService.kt
package no.novari.flyt.history.mapping.selectable

import no.novari.flyt.history.model.Selectable
import no.novari.flyt.history.model.instance.ActiveTimePeriod
import org.springframework.stereotype.Service

@Service
class ActiveTimePeriodSelectableMappingService {
    fun toSelectable(activeTimePeriod: ActiveTimePeriod): Selectable<String> {
        return Selectable(
            value = activeTimePeriod.name,
            label = getDisplayText(activeTimePeriod),
        )
    }

    private fun getDisplayText(activeTimePeriod: ActiveTimePeriod): String {
        return when (activeTimePeriod) {
            ActiveTimePeriod.TODAY -> "Denne dagen"
            ActiveTimePeriod.THIS_WEEK -> "Denne uka"
            ActiveTimePeriod.THIS_MONTH -> "Denne måneden"
            ActiveTimePeriod.THIS_YEAR -> "Dette året"
        }
    }
}
